molecularnodes/entities/molecule/base.py

```diff
@@ -60,15 +49,108 @@ def __init__(self, file_path: Union[str, Path, io.BytesIO]):
         file_path : Union[str, Path, io.BytesIO]
             The file path to the file which stores the atomic coordinates.
         """
-        super().__init__()
-        self._parse_filepath(file_path=file_path)
-        self.file: str
-        self.array: np.ndarray
-        self._frames_collection: str | None
+        self._frames_collection: str | None = None
+        self._code: str | None = None
         self._entity_type = EntityType.MOLECULE
+        super().__init__()
+        self._read(file_path)
+        self.atom_array: AtomArray | AtomArrayStack
+        self._create_object()
```

Comment:
I think you are using self.tree in other locations? Maybe it would be helpful to declare it here as well.

Reply:
whoops.  no you are using self.object